Userland/Libraries/LibJS/Runtime/SharedArrayBufferPrototype.cpp

```diff
@@ -106,7 +108,8 @@ JS_DEFINE_NATIVE_FUNCTION(SharedArrayBufferPrototype::slice)
     auto* new_array_buffer_object = static_cast<ArrayBuffer*>(new_array_buffer.ptr());
 
     // 17. If IsSharedArrayBuffer(new) is true, throw a TypeError exception.
-    // FIXME: Check for shared buffer
+    if (!array_buffer_object->is_shared_array_buffer())
+        return vm.throw_completion<TypeError>(ErrorType::ThisCannotBeArrayBuffer);
```

Comment:
```suggestion
    if (new_array_buffer_object->is_shared_array_buffer())
        return vm.throw_completion<TypeError>(ErrorType::ThisCannotBeSharedArrayBuffer);
```
Though the error message is not accurate in this case, as it's not concerning `this` but `new`.

Reply:
Did not go with your suggestion because the spec comment was incorrect, I double checked with [#sec-sharedarraybuffer.prototype.slice](https://tc39.es/ecma262/#sec-sharedarraybuffer.prototype.slice) and adapted the spec comment accordingly (throw if false, not throw if true).

Regarding the error message, I changed it to `ErrorType::SpeciesConstructorDidNotCreate`, is that more accurate? If not, what would be an accurate Error?